packages/react-native-sdk/src/components/Livestream/LivestreamTopView/DurationBadge.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { useTheme } from '../../../contexts';
+import { ShieldBadge } from '../../../icons';
+import { useCall } from '@stream-io/video-react-bindings';
+import { CallSessionResponse, StreamCallEvent } from '@stream-io/video-client';
+
+/**
+ * Props for the HostDurationBadge component.
+ */
+export type DurationBadgeProps = {
+  mode?: 'host' | 'viewer';
+};
+
+/**
+ * The HostDurationBadge component displays the duration while the live stream is active.
+ */
+export const DurationBadge = ({ mode }: DurationBadgeProps) => {
+  const [duration, setDuration] = useState(0);
+  const call = useCall();
+  const {
+    theme: {
+      colors,
+      variants: { iconSizes },
+      durationBadge,
+    },
+  } = useTheme();
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timer;
+    const handleLiveStarted = () => {
+      intervalId = setInterval(() => {
+        setDuration((d) => d + 1);
+      }, 1000);
+    };
+
+    const handleLiveEnded = (event: StreamCallEvent) => {
+      const callDetails = (
+        event as StreamCallEvent & {
+          call: { session: CallSessionResponse };
+        }
+      ).call.session;
+      if (callDetails?.live_ended_at !== null) {
+        clearInterval(intervalId);
+      }
+    };
+
+    if (mode === 'host') {
+      call?.on('call.live_started', handleLiveStarted);
+      call?.on('call.updated', handleLiveEnded);
+    } else {
+      handleLiveStarted();
+    }
+
+    return () => {
+      if (mode === 'host') {
+        call?.off('call.live_started', handleLiveStarted);
+        call?.off('call.updated', handleLiveEnded);
+      }
+      if (intervalId) {
+        clearInterval(intervalId);
+      }
+    };
+  }, [call, mode]);
+
+  const timestamp = new Date(duration * 1000).toISOString().slice(11, 19);
```

Comment:
Two things here:

1. `duration` itself should be in `ms`, instead of multiplying it by 1000 here.
2. Parsing approach is not good. Please do what react sdk has done - https://github.com/GetStream/stream-video-js/blob/74eeacfb02b3be88764ab2d005eade7cca90872e/packages/react-sdk/src/core/components/CallLayout/LivestreamLayout.tsx#L225

Reply:
You can just move `formatDuration` to bindings project.

Reply:
Done @vishalnarkhede